vcell-core/src/main/java/cbit/vcell/resource/NativeLib.java

```diff
@@ -69,4 +75,37 @@ public boolean isDone( ) {
 		return NativeLoader.load(libName).isDone();
 	}
 
+	private String generatePath(){
+		OperatingSystemInfo osi = OperatingSystemInfo.getInstance();
+		String installDir = ResourceUtil.getVCellInstall().getAbsolutePath();
+		String nativeLibDir = "nativelibs";
+		String osDir = osi.getNativeLibDirectory();
+		String libFileName = this.getLibraryFileName();
+
+		Path path = Paths.get(installDir, nativeLibDir, osDir, libFileName);
+		return path.toAbsolutePath().toString();
+	}
+
+	private String getLibraryFileName(){
+		OperatingSystemInfo osi = OperatingSystemInfo.getInstance();
+		String fileName = osi.isWindows() ? "" : "lib" ; // unix library start with lib
+		fileName += this.libName;
+		return fileName + "." + this.getAppropraiteLibrarySuffix();
+	}
+
+	private String getAppropraiteLibrarySuffix(){
+		String suff;
+		OperatingSystemInfo osi = OperatingSystemInfo.getInstance();
+		if (osi.isWindows()){
+			suff = "dll";
+		} else if (osi.isMac()){
+			suff = "jnilib";

```

Comment:
dylib is the modern suffix for mac ... we don't have any because our libraries are old.   Maybe it makes sense to change this to 'getAppropriateLibrarySuffixes()' and return new String[] { "jnilib", "dylib" }